<?xml version="1.0"?>
<launch>
  <arg name="base_frame" default="base_link"/>
  <arg name="tool_frame" default="tool0"/>
  <arg name="save_path" default="/tmp/calibration"/>
  <arg name="config_file" default="$(find industrial_calibration_ros)/config/target_detector_config.yaml"/>
  <arg name="sync_time" default="1.0"/>

  <!-- Target detector node -->
  <node name="target_detector_node" pkg="industrial_calibration_ros" type="industrial_calibration_ros_target_detector" output="screen">
    <param name="config_file" value="$(arg config_file)"/>
  </node>

  <!-- Data collector -->
  <remap from="image" to="image_detected"/>
  <node name="data_collection_3d_node" pkg="industrial_calibration_ros" type="3d_data_collection_node.py" output="screen">
    <param name="base_frame" value="$(arg base_frame)"/>
    <param name="tool_frame" value="$(arg tool_frame)"/>
    <param name="save_path" value="$(arg save_path)"/>
    <param name="sync_time" value="$(arg sync_time)"/>
  </node>
</launch>
